mesa/agent.py

```diff
@@ -264,17 +264,25 @@ def do(
 
         return res if return_results else self
 
-    def get(self, attr_name: str) -> list[Any]:
+    def get(self, attr_names: str | list[str]) -> list[Any]:
         """
-        Retrieve a specified attribute from each agent in the AgentSet.
+        Retrieve the specified attribute(s) from each agent in the AgentSet.
 
         Args:
-            attr_name (str): The name of the attribute to retrieve from each agent.
+            attr_names (str | List[str]): The name(s) of the attribute(s) to retrieve from each agent.
 
         Returns:
-            list[Any]: A list of attribute values from each agent in the set.
+            list[Any]: A list with the attribute value for each agent in the set if attr_names is a str
+            List[List[Any]]: A list with a list of attribute values for each agent in the set if attr_names is a list of str
         """
-        return [getattr(agent, attr_name) for agent in self._agents]
+
+        if isinstance(attr_names, str):
+            return [getattr(agent, attr_names) for agent in self._agents]
```

Comment:
Maybe we should default to None (getattr(agent, attr_names, None)). This way it would be easier to collect across different agent types, which might or might not have the specified attributes.

Reply:
I see your point but I am torn. I personally probably would prefer getting the error explicitly rather than being confronted with None values downstream.

Reply:
Past discussion: https://github.com/projectmesa/mesa-examples/pull/90. The summary is that it depends on the coder's background: in JavaScript, it is not surprising for `obj.attr` to return `undefined` and not raise an error. However, Python doesn't distinguish between `null` and `undefined`, and so there is an information loss when returning `None`. It is ambiguous and could mean the modeler intentionally chooses to return `None` instead of an absence of attribute.
I prefer no fallback to `None`.

Reply:
Maybe we could add a boolean switch (something like `allow_undefined`) that defaults to `False`. When `False` it isn't allowed and an error is returned, when True it is allowed and `None` values are returned.

Reply:
that would make the code internally more complicated. You would get two additional nested if/else. 

The easier solution is for the user to do agentset.select("condition to ensure attribute exists").get(). 

Also, why allow None in the first place? It only gives more downstream problems.

Reply:
I think its not necesarily an "error" if we collect None instead of triggering an AttributeError. It depends on the use case. If you quickly want to get some attributes on all of your agents, but some agents don't have that attribute you still get some result. Otherwise you first have to pre-select the right AgentSet and then collect the attributes.

I think the cleanest solution would probably be to mimic getattr in that users can provide a default value for `agentset.get()` and pass that to getattr, if set.

Reply:
implementation wise this is a bit tricky. `default` in `getattr` is not a keyword argument but an argument. So you cannot use `None` to signal you don't specify default.

Reply:
We can defer this discussion and implementation in a separate PR. The current behavior in this PR is consistent with the existing API, which does not fall back to `undefined`/`None`.